src/modules/context2d.js

```diff
@@ -1341,6 +1346,20 @@
         } else {
             this.pdf.addImage(img, 'JPEG', xRect.x, xRect.y, xRect.w, xRect.h, null, null, angle);
         }
+        
+        function crop(canvas, offsetX, offsetY, width, height) {
+            // create an in-memory canvas
+            var buffer = document.createElement('canvas');
+            var b_ctx = buffer.getContext('2d');
+            // set its width/height to the required ones
+            buffer.width = width;
+            buffer.height = height;
+            // draw the main canvas on our buffer one
+            b_ctx.drawImage(canvas, offsetX, offsetY, width, height, 0, 0, buffer.width, buffer.height);
```

Comment:
If you do it like this, you have dependency on canvas element. Also I dont want to have more images in the pdf, if we can achieve the same with clipping.